plugins/defaultplugins/l3plugin/route_config.go

```diff
@@ -75,104 +62,91 @@ func (plugin *RouteConfigurator) Init() (err error) {
 }
 
 // ConfigureRoutes process the NB config and propagates it to bin api calls
-func (plugin *RouteConfigurator) ConfigureRoutes(config *l3.StaticRoutes) (wasError error) {
-	routes := plugin.protoRoutesToStruct(config)
-
-	for i := range routes {
-		err := plugin.vppAddRoute(routes[i])
-		if err != nil {
-			wasError = err
+func (plugin *RouteConfigurator) ConfigureRoutes(config *l3.StaticRoutes_Route) error {
+	log.Infof("Creating new route set with destination address %v", config.DestinationAddress)
+	routes, err := plugin.transformRoute(config)
+	if err != nil {
+		return err
+	}
+	if len(routes) > 0 {
+		for _, route := range routes {
+			err := plugin.vppAddRoute(route)
+			if err != nil {
+				return err
+			}
+			routeIdentifier := routeIdentifier(route.DstAddr.String(), route.NextHop.Addr.String())
+			plugin.RouteIndexes.RegisterName(routeIdentifier, plugin.RouteIndexSeq, nil)
+			plugin.RouteIndexSeq++
+			log.Infof("Route %v registered", routeIdentifier)
 		}
 	}
 
-	return wasError
+	return nil
 }
 
-// ModifyRoutes process the NB config and propagates it to bin api calls
-func (plugin *RouteConfigurator) ModifyRoutes(newConfig *l3.StaticRoutes, oldConfig *l3.StaticRoutes) (
-	wasError error) {
-	newRoutes := plugin.protoRoutesToStruct(newConfig)
-	oldRoutes := plugin.protoRoutesToStruct(oldConfig)
-	toBeDeleted, toBeAdded := plugin.diffRoutes(newRoutes, oldRoutes)
-	for i := range toBeDeleted {
-		err := plugin.vppDelRoute(toBeDeleted[i])
+// ModifyRoute process the NB config and propagates it to bin api calls
+func (plugin *RouteConfigurator) ModifyRoute(newConfig *l3.StaticRoutes_Route, oldConfig *l3.StaticRoutes_Route) error {
+	log.Infof("Modifying route set with destination address %v", oldConfig.DestinationAddress)
+	newRoutes, err := plugin.transformRoute(newConfig)
+	if err != nil {
+		return err
+	}
+	oldRoutes, err := plugin.transformRoute(oldConfig)
+	if err != nil {
+		return err
+	}
+
+	for _, oldRoute := range oldRoutes {
+		err := plugin.vppDelRoute(oldRoute)
```

Comment:
Calculate the diffrence. Do not do delete all add existing (do not thouch not changed...)